### ISSUE TITLE
Dynamic Labels for Torrent Downloads

### ISSUE THREAD

**Issue by chach17** [Sonarr/Sonarr#2163]
Currently when configuring a Torrent based download client, there is the option to add a Label to the torrent download client.  The label can be quite useful, not only for categorizing downloads, but also because further automation can be driven off of the label after the download has completed, such as for rules based auto-removal of torrents from the download client.  Currently any kind of Label based automation can only be driven off a single static label that can be used to separate Sonarr downloads from others, but it could be so much more useful if the label could be dynamic so that automation could be driven at the series, season, or even indexer levels.

The feature request is to change the "Label" text field in the settings for a specific Downloader to a dropdown box with the following selections:

1. Custom Label.  When this is chosen a new text field becomes visible for the user to enter a static label as is currently available.
2. Indexer.  The name of the indexer that found the torrent.  Perhaps this could also be specified in the settings for a particular Indexer and then just applied to the torrent download as the Label.
3. Series Title
4. Custom Format.  I'm thinking something along the lines of how format specifiers work with printf.  The idea is you can specify your own format and specific pieces of information, such as series name, title, or episode number can be substituted into the format string.  So if I wanted a label that consisted of the the series name and the indexer that found the torrent, then perhaps a format string would be:
`%series_title% - %indexer_name%`

and this might result in a label of:

`The Best Show Ever - ThePirateBay`

**Comment by markus101** [comment]
What are you wanting to automate off the labels?

The problem with this is it would only work for clients that support, both multiple labels and dynamic labels (they don't need to be created before they can used). Without the ability to specify multiple labels Sonarr would be unable to track and filter downloads via a label for it's use (the primary reason we recommend using a label) and if they can't be dynamically added anything other than a single static label would only work if you manually pre-filled them in.

We'd need to know which clients support multiple labels, to determine if it's worth the effort, but at the moment I'm leaning towards this being something we wouldn't implement, and if we did, not something that would be high priority for us to implement.

**Comment by jonnes86** [comment]
I use uTorrent and this feature would be amazing. Right now sorting all of the different trackers that have different seeding requirements is incredibly tedious and time consuming. If I could label based off of the indexer I would be able to sort by that field in uTorrent and easily manage my downloads.

**Comment by LinusT** [comment]
This is somewhat related to my recent request #2469, as far as I want to be able to change label (no need for multiple ones) after successful (or failed) import by Sonarr. This allows simple scripts to run in client to prune according to need.

**Comment by daniel-loader** [comment]
In my use case I'd love to be able to differentiate between a tv episode and tv season pack on the deluge side of things as they often have different seeding rules. Since sonarr anlready has season pack detection I'd hope that isn't hugely out of scope here.

**Comment by valterm** [comment]
I'd like to double down on the deluge side of things: with the Labels plugin custom rules can be set for different labels, and it would make automating the handling of completed downloads easier. Primarily, it would eliminate the need to copy or hardlink the files after finishing the download - episodes for a show could be labeled with the show's name, and the finished download could be moved by deluge itself into the new location. It would allow to continue seeding without having to use workarounds.

**Comment by devolution2409** [comment]
sonarr-imported label would be a huge improvement as it would allow autoremoveplus plugin (or really, any plugin that does the same job) removing the torrent + data (since you can hardlink the file with sonarr) only if the episode has been imported.

A label based on indexer name would be cool aswell since you could exempt them from the remove rule. (ie tracker that have a ratio rule)

**Comment by grozan** [comment]
the issue #3537 I opened to propose essentially the same thing was closed, marked as a duplicate of that one, that I had missed. Sorry for the dupe.
Seeing that this one is opened for more than 2 years actually scares me :-(
Apparently there are users interested in this for their custom workflows.
It probably would not be used by most people, and wouldn't work out of the box with every download client out there. I get that.
Still, it would unlock new possibilities for some without affecting the others, so couldn't this be implemented and communicated as a "use it at your own risk" kind of thing?

Can we at least quickly get a decision on that one?
Thanks in advance
